azure_functions_worker/bindings/meta.py

```diff
@@ -94,5 +107,54 @@ def to_outgoing_proto(binding: str, obj: typing.Any, *,
             f'unable to encode outgoing TypedData: '
             f'unsupported type "{binding}" for '
             f'Python type "{type(obj).__name__}"')
+    return datum
 
-    return datumdef.datum_as_proto(datum)
+
+def to_outgoing_proto(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type],
+                      shmem_mgr: SharedMemoryManager,
+                      invocation_id: str) -> protos.TypedData:
+    datum = get_datum(binding, obj, pytype)
+    return datumdef.datum_as_proto(datum, shmem_mgr, invocation_id)
+
+
+def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type],
+                      out_name: str,
+                      shmem_mgr: SharedMemoryManager,
+                      invocation_id: str) -> protos.ParameterBinding:
+    datum = get_datum(binding, obj, pytype)
+    # TODO gochaudh: IMPORTANT: Right now we set the AppSetting to disable this
+    # However that takes impact only for data coming from host -> worker
```

Comment:
Would it be possible to reply with shared memory only if the input was shared memory?

Reply:
That would not handle cases where if the function doesn't take any input (say just an HTTP invocation) and produces a large output that needs to go back from worker to host over shared memory. What we need is a way to enable and disable both the worker and host with the same switch (which I think is going to be an environment variable or some App Settings thingy - will check with folks when we release this PR.)